feat: add incident statistics child API definition

tsedekey opened the pull request:
## Description

<!-- Describe the goal and purpose of this PR. -->
This PR introduces the REST API definition for the child Incident Statistics endpoint. It defines the contract for retrieving aggregated incident statistics for a given error hash code.

## Checklist

<!--- Please delete options that are not relevant. Boxes should be checked by reviewer. -->
- [ ] Enable backports when necessary (fex. [for bug fixes](https://github.com/camunda/camunda/blob/main/CONTRIBUTING.md#backporting-changes) or [for CI changes](https://github.com/camunda/camunda/wiki/CI-&-Automation#when-to-backport-ci-changes)).

## Related issues

closes #42649

tsedekey committed:
feat: add API  definition for endpoint

fix: incorrect name used

tsedekey committed:
feat: add Java client definition

refactor: addres CP PR comments

tmetzke commented on zeebe/gateway-protocol/src/main/proto/v2/incidents.yaml:
```diff
@@ -109,6 +109,44 @@ paths:
+      description: |
+        Returns incident statistics by error hash code.
+      parameters:
+        - name: errorHashCode
```
💭 I am wondering if this should rather be a mandatory parameter in the request body. We are not identifying a resource with the hash. But path parameters are exactly meant for that. The same then goes for PD process instance statistics, to be honest.

tsedekey replied:
Thanks for the explanation @tmetzke. I’ve taken this on board and agree that using a dynamic path segment for the error hash code is not ideal here. To avoid ambiguity and keep the API structure explicit, I’m proposing to change the child endpoint to a dedicated URL:
```code
/incidents/statistics/process-instances/error-hash-code
```
The error hash code would then be provided as a required field in the request body filter (as an integer), rather than as a path parameter. This keeps the parent/child statistics endpoints clearly separated at the URL level, avoids collisions with other incident paths, and still makes the scoping dimension explicit and mandatory.

Let me know if this direction works for you!

tmetzke reviewed (commented):
👍 I like the direction and general API definition.
💭 Similar to the parent endpoint, I propose to align this better with other statistics endpoints.
💭 I believe removing the "child" differentiator from the path for statistics is more future-proof and more RESTful. We can create a follow-up for PD process instance statistics as well if we want to go that route.

tsedekey committed:
feat: add rest controller implementation

refactor: rename to incident error hash code PI statistics

refactor: add pd id

test: fix test url

test: fix failing test

refactor: change error hash code to integer

refactor: change error hash code to integer 2

refactor: rename and include error hash code in filter

refactor: combine tests

refactor: address CP PR comments

refactor: address CP PR comments 2

refactor: address CP PR comments 2

fix: filter consumers

refactor: address CP PR comments 3

style: fix spotless errors

refactor: address PR comments

test: update assert parameter

docs: shorten summary

Co-authored-by: Tobias Metzke-Bernstein <586643+tmetzke@users.noreply.github.com>

docs: shorten operationId

Co-authored-by: Tobias Metzke-Bernstein <586643+tmetzke@users.noreply.github.com>

docs: alphabetically sort sorting fields

Co-authored-by: Tobias Metzke-Bernstein <586643+tmetzke@users.noreply.github.com>

docs: include errorHashCode parameter in method docs

Co-authored-by: Tobias Metzke-Bernstein <586643+tmetzke@users.noreply.github.com>

refactor: re-use provideSearchRequestProperty

Co-authored-by: Tobias Metzke-Bernstein <586643+tmetzke@users.noreply.github.com>

tsedekey commented:
@tmetzke thanks for the in-depth and very helpful comments — I’ve gone through them and (hopefully) addressed everything now ✅

tmetzke commented:
I think you need to rebase due to some other PR having added API to the client 🙈

tmetzke reviewed (commented):
👍 API spec looks great, looking at the rest now.

tmetzke commented on zeebe/gateway-rest/src/test/java/io/camunda/zeebe/gateway/rest/controller/IncidentControllerTest.java:
```diff
@@ -43,6 +46,26 @@ public class IncidentControllerTest extends RestControllerTest {
   static final String INCIDENT_SEARCH_URL = INCIDENT_BASE_URL + "/search";
   static final String INCIDENT_PROCESS_INSTANCE_STATISTICS_URL =
       INCIDENT_BASE_URL + "/statistics/process-instances";
+  static final String INCIDENT_STATISTICS_URL =
```
🙃 I guess that constant's name could be more precise, maybe. It's process instance statistics by definition, I guess. 🙂

tmetzke commented on zeebe/gateway-rest/src/test/java/io/camunda/zeebe/gateway/rest/controller/IncidentControllerTest.java:
```diff
@@ -169,4 +193,195 @@ void shouldReturnIncidentProcessInstanceStatistics() {
+  }
+
+  @Test
+  void shouldRejectIncidentProcessInstanceStatisticsByDefinitionRequestWithoutFilter() {
```
🔧 We can also test the empty `filter` attribute in the request body that should fail.

tmetzke commented on clients/java/src/main/java/io/camunda/client/CamundaClient.java:
```diff
@@ -3123,4 +3124,29 @@ IncidentsByElementInstanceSearchRequest newIncidentsByElementInstanceSearchReque
+   * <pre>
+   * camundaClient
+   *  .newIncidentProcessInstanceStatisticsByDefinitionRequest()
+   *  .filter(f -> f.errorHashCode(123456789))
```
🔧 The error message hash is required at the moment, so the client could also enforce this by either the method signature (containing the hash code parameter) or by the stepped builder pattern (making the hash code the only first option in the interface before the next steps are allowed). The latter is more complex but also more flexible. We might not need this flexibility here, though.

tmetzke reviewed (commented):
👍 Looks really good overall, thanks, @tsedekey 🚀 
🔧 Let's reflect that the hash code is required in the client as well.

tmetzke reviewed (approved):
👍 Looks good to me, nice job! 🚀 
🔧 I've added some smaller improvement ideas, nothing blocking, though. Please look into them but feel free to merge when they're addressed 🙂

tsedekey committed:
fix: add missing import
